### DISCUSSION TITLE
Definitions and Realizable Entities in BFO

### DISCUSSION THREAD

**Discussion by gregfowlerphd** [JisooSeo/101-WG-Discussions#10] [General]
I've recently been reading some papers by Barry to deepen my understanding of BFO, and I've noticed some problems with the definitions given in those papers, particularly when the definitions involve realizable entities.

Here's one example. A paper entitled "A First-Order Logic Formalization of the Industrial Ontologies Foundry Signature Using Basic Formal Ontology" gives the following definition of "manufacturing tool" (p. 6):

manufacturing tool(x) <-> material-entity(x) & Ef(BFO:function(f) & has-function(x, f) & Ay(realizes(y, f) -> manufacturing process(y)))

This has pretty clearly got to be wrong, I think: Given this definition, any material entity that has a function that it never realizes--say a kitchen knife that is never used to cut anything--trivially counts as a manufacturing tool.

Here's another, albeit less egregious, example. In a paper entitled "Ontology of Plays for Autonomous Teaming and Collaboration", the following definition is given of "authority role" (p. 15 of the proceedings the paper was published in; p. 7 of the PDF downloaded from PhilPapers):

CCO: Authority Role =def a Role that is realized by Acts which create, modify, transfer, or eliminate Action Regulations or other Authority Roles, and inheres in an Agent in virtue of collective acceptance of that Agent's ability to issue binding directives.

(Presumably this definition comes straight from CCO.) Given this definition, it would seem that there can be no unrealized authority roles (since it states that an authority role is one that *is realized* by certain acts), which certainly seems mistaken.

(Note re: the second definition: It may be that the logical definition and/or axioms associated with "authority role" in CCO do not have the problematic implication noted.)

Anyone else have any thoughts on these examples? And please feel free to tell me that I'm completely off-base if I am. :)

**Comment by JisooSeo** [comment]
These are very interesting observations! 
For the definition of "manufacturing tool," I would say a kitchen knife is not a manufacturing tool, according to the definition, if it is not realized by a manufacturing process. It is clear in the last conjunct ("Ay(realizes(y, f) -> manufacturing process(y))") that anything that realizes the function of x is in fact a manufacturing process. 
What I have qualms about the definition is that anything can be a manufacturing process if it "realizes" the function. This seems to require more explanation.

**Reply by gregfowlerphd** [reply]
Thanks for your response, Jisoo. :)

In the case I'm considering, the kitchen knife is never used to cut anything, so its function, which we can call c, is never realized. Hence, everything that realizes c--all none of them--is a manufacturing process. So the last conjunct of the definition is satisfied where f=c.

Here's another way of thinking about it. Consider the conditional embedded in the last conjunct of the definition. Since c is never realized, for all y, the antecedent of that conditional is false (where f=c). But whenever the antecedent of a material conditional is false, the whole conditional is true. Thus, for all y, the whole conditional embedded in the last conjunct of the definition is true, which is just what the last conjunct says. Therefore, the last conjunct of the definition is satisfied where f=c.

**Reply by JisooSeo** [reply]
I see! The last conjunct is trivially true therefore rendering the main biconditional true. 
Relatedly, would there be a possibility where the unused kitchen knife would be realized, even in a trivial way, but whatever realizes it is not the manufacturing process? (I guess it depends on how "manufacturing process" is defined.)

**Reply by gregfowlerphd** [reply]
Yep, you got it! :)

I assume that when you say "where the unused kitchen knife would be realized" you mean "where the unused kitchen knife's *function* would be realized"? After all, it's functions (and other dispositions) that are realized in BFO, not material entities like kitchen knives.

If so, I certainly think there's a possibility that the unused kitchen knife's function--to cut things, assuming that's its only function--would be realized but not realized by a manufacturing process. (Of course, in that possibility, it wouldn't be an *unused* kitchen knife.) Indeed, that would seem to be the standard case: The process of a kitchen knife cutting something is not typically, if ever, a manufacturing process.

I doubt, however, that in the possibility under discussion, the definition from Barry's paper would count the kitchen knife as a manufacturing tool, in which case it couldn't serve as a counterexample to that definition. But perhaps you're suggesting that maybe there's a way for the kitchen knife's function to be realized in a process that isn't a manufacturing process and yet for the kitchen knife to still count as a counterexample to the definition of "manufacturing tool"?

**Reply by JisooSeo** [reply]
Yeah, I think I'm trying to think of a counterexample of such kind in which the function of a thing is realized but not by a manufacturing process. Let's say I use a kitchen knife to only scratch my back and for nothing else. It seems like some function of the knife is "realized" but still, it is not realized by a manufacturing process. Does it make sense to say that the kitchen knife is not a manufacturing tool? Intuitively, I would call it a manufacturing tool, against what the definition would define it.

And come to think of it, maybe it is not so problematic that an unused kitchen knife is (trivially) categorized as a manufacturing tool. As long as something in fact has a function that _can_ be realized, shouldn't it be considered a manufacturing tool?

**Reply by gregfowlerphd** [reply]
Well, it was supposed to be problematic that an unused kitchen knife is categorized as a manufacturing tool because it was supposed to be intuitive that kitchen knives aren't manufacturing tools. But it sounds like maybe you don't share that intuition (?).

In that case, do you share the intuition that, say, baby monitors aren't manufacturing tools? Or that action figures aren't? Because similar counterexamples to the definition of "manufacturing tool" can be developed using baby monitors and action figures. Just take a baby monitor or action figure that never realizes its function, and according to the definition of "manufacturing tool", that baby monitor or action figure will count as a manufacturing tool, even though it seems that it shouldn't.

**Reply by gregfowlerphd** [reply]
Jisoo, you said in your most recent reply:

"Let's say I use a kitchen knife to only scratch my back and for nothing else. It seems like some function of the knife is 'realized' but still, it is not realized by a manufacturing process."

Sorry I didn't address this remark earlier, but I needed to grab dinner, so I focused on the issue of constructing a counterexample along the lines intended in my original post. Now that I have time, however, I should mention that I *think* that (on the BFO understanding of functions) the knife doesn't realize one of its functions when you use it to scratch your back, because (roughly) it wasn't designed with back-scratching in mind.

Barry's paper "Functions in Basic Formal Ontology" might be helpful here: https://philpapers.org/archive/SPEFIB.pdf. It certainly helped me better understand BFO functions.

**Reply by JisooSeo** [reply]
Yeah, I think that "the knife doesn't realize one of its functions when you use it to scratch your back, because (roughly) it wasn't designed with back-scratching in mind" as well, but then the definition should indicate that the function has to align with the purpose of the design, but I guess I don't know the definition of "function" in BFO, 

I was thinking of changing the last conjunct Ay(realizes(y, f) -> manufacturing process(y)) into Ey(manufacturing process(y) & realizes(y, f)). What do you think?

**Reply by JisooSeo** [reply]
Thanks for the link! I will go ahead and read it :)

**Reply by gregfowlerphd** [reply]
>> Thanks for the link! I will go ahead and read it :)

No problem. Hope you enjoy! :)

>> the definition should indicate that the function has to align with the purpose of the design

So I *think* that the alignment in question is supposed to be built into the BFO notion of function, in which case I don’t think it needs to be reiterated in the definition. The paper I linked to might help here.

>> I was thinking of changing the last conjunct Ay(realizes(y, f) -> manufacturing process(y)) into Ey(manufacturing process(y) & realizes(y, f)). What do you think?

Well, I think that the original definition was supposed to allow for the possibility of manufacturing tools that are never used in manufacturing processes, which your suggested change would rule out. Not only does it seem correct, from a philosophical perspective, that lack of use is consistent with being a manufacturing tool, I also suspect that users of the Industrial Ontologies Foundry wouldn’t want to rule it out. Suppose that at one of their factories, a IOF user has a machine press that is never used. What ontological category are they to employ in classifying it if not manufacturing tool?

If you want my honest opinion, I don’t think you can formulate an adequate definition of “manufacturing tool” along the lines of the original using only classical first-order logic. You have to bring in modal notions. Thus, replacing the last conjunct with either of the following would be a huge improvement from a definitional perspective:

if it were the case that Ey(realizes(y, f), then it would be the case that Ey(manufacturing process(y) & realizes(y, f))

OR

Necessarily, (Ay(realizes(y, f) -> manufacturing process(y))

Unfortunately, while I don’t think there’s anything in BFO itself that proscribes using modal notions, my understanding is that neither OWL nor the reasoners used with ontologies can handle them.

**Comment by JisooSeo** [comment]
> Well, it was supposed to be problematic that an unused kitchen knife is categorized as a manufacturing tool because it was supposed to be intuitive that kitchen knives aren't manufacturing tools. But it sounds like maybe you don't share that intuition (?).
>
Oh, I see. Yeah, I was wondering about that as well. I think a kitchen knife can be a manufacturing tool (if "manufacture" is understood in a broad term) but I guess it's not a good definition if it depends too much on intuitions about whether a thing is a tool or not. 
> 
> In that case, do you share the intuition that, say, baby monitors aren't manufacturing tools? Or that action figures aren't? Because similar counterexamples to the definition of "manufacturing tool" can be developed using baby monitors and action figures. Just take a baby monitor or action figure that never realizes its function, and according to the definition of "manufacturing tool", that baby monitor or action figure will count as a manufacturing tool, even though it seems that it shouldn't.

I accept the counterexamples, but I think they're the result of the ambiguity of the last conjunct. If a baby monitor is used in nailing some nails into a wall so its function is realized in some way, then, according to the definition, it is a manufacturing tool which is counterintuitive. I think the reason we think a baby monitor can be a counterexample is that we already have a definition of a baby monitor, an entity that has the function of monitoring babies. So it seems like what a "manufacturing tool" or not is already embedded in the conceptualization of a baby monitor which sounds circular.
